hnn_core/network.py

```diff
@@ -394,12 +409,76 @@ def __init__(self, params, add_drives_from_params=False,
     def __repr__(self):
         class_name = self.__class__.__name__
         s = ("%d x %d Pyramidal cells (L2, L5)"
-             % (self._params['N_pyr_x'], self._params['N_pyr_y']))
+             % (self._N_pyr_x, self._N_pyr_y))
         s += ("\n%d L2 basket cells\n%d L5 basket cells"
               % (len(self.pos_dict['L2_basket']),
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
+    def set_cell_positions(self, *, n_pyr_x=None, n_pyr_y=None,
```

Comment:
thinking a bit more about this don't think `n_pyr_x` or `n_pyr_y` should be exposed here. Because `n_pyr_x` and `n_pyr_y` also dictate the number of cells. Once you have created cells using [`_add_cell_type`](https://github.com/jonescompneurolab/hnn-core/blob/d3e51aee3f4a2aea63b411448f1aea8fa80ae44f/hnn_core/network.py#L382-L383), it should not be possible to change the number of cells. I think we can expose these arguments as part of `add_cell_type` when we make it public. The `pos_dict` would get created inside that method. Then the `repr` of the `Network` class will tell you how many cells per layer per cell type and if users are more curious they should inspect `net.pos_dict` or the visualization.

A second issue is that this will lead to inconsistencies in the API. For e.g., why would you want to control number of pyramidal cells but not the number basket cells?